packages/flutter/lib/src/gestures/binding.dart

```diff
@@ -78,9 +79,28 @@ mixin GestureBinding on BindingBase implements HitTestable, HitTestDispatcher, H
   static GestureBinding get instance => _instance;
   static GestureBinding _instance;
 
+  /// Determines whether the binding will record the input [ui.PointerData] to
+  /// the timeline.
+  ///
+  /// This flag is bypassed in release mode.
+  ///
+  /// See [serializePointerData] and [deserializePointerData].
+  bool recordInput = false;
+
   final Queue<PointerEvent> _pendingPointerEvents = Queue<PointerEvent>();
 
   void _handlePointerDataPacket(ui.PointerDataPacket packet) {
+    if (!kReleaseMode && recordInput) {
+      Timeline.instantSync(
+        'GestureBinding receive PointerEvents',
+        arguments: <String, List<Map<String, dynamic>>>{
+          'events': <Map<String, dynamic>>[
+            for(final ui.PointerData datum in packet.data)
+              serializePointerData(datum),
+          ],
+        },
+      );
```

Comment:
This will be slow enough to significantly disrupt benchmark measurement.

Do you really need all this data for every event? I suspect even recording a single datapoint from this this frequently could throw things off, but recording the whole packet every time definitely will.

Reply:
At the very least, we should document that this is expensive and will throw off various measurements in the timeline.

Reply:
I'm not expecting to do this in a real benchmark test but to generate data for benchmark tests. But absolutely I need to document it to point out it's very slow.

Reply:
I put it in an assert. Hope this can fix the concern.